doc/check.texi

```diff
@@ -1032,6 +1032,51 @@ respectively.
 Similar to @code{ck_assert_int_*}, but compares two unsigned integer values
 (@code{uintmax_t}) instead.
 
+@item ck_assert_floating_eq
+@itemx ck_assert_floating_ne
+@itemx ck_assert_floating_lt
+@itemx ck_assert_floating_le
+@itemx ck_assert_floating_gt
+@itemx ck_assert_floating_ge
+
+Compares two floating point numbers (@code{float}, @code{double},
+@code{long double}) and displays predefined message with condition and
+values of both input parameters on failure.  The operator used for comparison
+is different for each function and is indicated by the last two letters
+of the function name.  The abbreviations @code{eq}, @code{ne}, @code{lt},
+@code{le}, @code{gt}, and @code{ge} correspond to @code{==}, @code{!=},
+@code{<}, @code{<=}, @code{>}, and @code{>=} respectively. Beware
+using those operators for floating point numbers because of precision
+possible loss on every operation on floating point numbers. For example
+(1/3)*3==1 would return false, because 1/3==1.(3) and cannot be represented
+by computer logic. As another examples 1.1f in fact could be
+1.10000002384185791015625 and 2.1f could be 2.099999904632568359375 because of
+binary representation of floating point numbers.
+Consider using precision comparisons or integer numbers instead.
+
+@item ck_assert_floating_eq_tol10
+@itemx ck_assert_floating_ne_tol10
+@itemx ck_assert_floating_lt_tol10
+@itemx ck_assert_floating_le_tol10
+@itemx ck_assert_floating_gt_tol10
+@itemx ck_assert_floating_ge_tol10
+
+Compares two floating point numbers (@code{float}, @code{double},
+@code{long double}) with specified tolerance (power of 10) set by the third
+parameter (@code{unsigned char}) and displays predefined message with condition
+and values of both input parameters and precision on failure. The abbreviations
+@code{eq}, @code{ne}, @code{lt}, @code{le}, @code{gt}, and @code{ge} correspond
+to @code{==}, @code{!=}, @code{<}, @code{<=}, @code{>}, and @code{>=}
+respectively with acceptable error (tolerance) specified by the last parameter.
+Beware of using those operators for floating operators because of unclear bounds
+of exponent and because floating point errors are platform dependant. Floating
+point numbers often internally represented as binary so they can't be exactly
+power of 10. All these operators have significant error in comparison so use
+them only if you know what you're doing. As axample
+@code{ck_assert_floating_le_tol10(0.002*10, 0.001*10, 2)} would fail
```

Comment:
Here is the answer:
[http://stackoverflow.com/questions/18494237/floating-point-mismatch-between-compilers-visual-studio-2010-and-gcc](http://stackoverflow.com/questions/18494237/floating-point-mismatch-between-compilers-visual-studio-2010-and-gcc)
I've figured it out on your ms platform tests while testing bounds. GCC and clang were passed. But one of compilers didn't passed test. So I had to remove testing bounds.

Moreover I''ll remove ge_tol10 and le_tol10. They could not give the same result on some values on different platforms and in fact they break logic of comparing with specified precision because values 0.01 and 0.02 could become equal with tolerance pow(10, -2) in case of <=. With gt operator they wouldn't.